### PR TITLE
[Chore] Fix issue when security is disabled and testing from WS Console

### PR DESCRIPTION
**Pull request by Thushani-Jayasekera**
### Issues

Fixes https://github.com/wso2-enterprise/choreo/issues/30190
Relates to https://github.com/wso2-enterprise/choreo/issues/26321

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/Utils.java [hunk 1]
```diff
@@ -90,6 +91,11 @@ static void populateRemoveAndProtectedHeaders(RequestContext requestContext) {
         // generic OPTIONS method call happens, matchedResourcePath becomes null.
         if (requestContext.getMatchedResourcePath() != null &&
                 requestContext.getMatchedResourcePath().isDisableSecurity()) {
+            // If the resource has disabled security, and  if the connection request
+            // is from the Choreo-console Test console UI,  it is required to add
+            // choreo-internal-API-Key header to the response header x-websocket-protocol
+            InternalAPIKeyUtils.addWSProtocolResponseHeaderIfRequired(requestContext);
```
Comment (renuka-fernando) [inline enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/Utils.java:97]: This function `populateRemoveAndProtectedHeaders` prevents protected headers from being passed to the backend. Shall we move this function call to the method `authenticate` in the class `UnsecuredAPIAuthenticator`?

Comment (renuka-fernando) [reply enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/Utils.java:97]: And shall we check for the API type? WS or Rest API

**Review by renuka-fernando — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[]  : dev-deployment-v2 : 20240825.2

**Comment by choreo-cicd** [comment]
[] Dataplane(EastUS) cluster : dev-deployment-v2 : 20240825.2

**Comment by choreo-cicd** [comment]
[] Dataplane(NorthEU) cluster : dev-deployment-v2 : 20240825.2

**Comment by choreo-cicd** [comment]
[] Dataplane(CDP2-WU2) cluster : stage-deployment-v2 : 20240826.1

**Comment by choreo-cicd** [comment]
[] Dataplane(EastUS) cluster : stage-deployment-v2 : 20240826.1

**Comment by choreo-cicd** [comment]
[] Controlplane cluster : stage-deployment-v2 : 20240826.1

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20240827.1

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(CDP2-WU2) cluster : prod-deployment-v2 : 20240827.1

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : prod-deployment-v2 : 20240827.1

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : prod-deployment-v2 : 20240827.1

**Pull request merged**